[0.16] Backports

fanquake opened the pull request:
Backports:
- #12626 Limit the number of IPs addrman learns from each DNS seeder
- #12650 gui: Fix issue: "default port not shown correctly in settings dialog"
- #12756 [config] Remove blockmaxsize option
- #12985 Windows: Avoid launching as admin when NSIS installer ends.
- #12946 depends: Fix Qt build with XCode 9.3
- #12998 Default to defining endian-conversion DECLs in compat w/o config
- #12999 qt: Show the Window when double clicking the taskbar icon
- #13064 List support for BIP173 in bips.md

to the 0.16 branch.

EthanHeilman committed:
Limit the number of IPs we use from each DNS seeder

A risk exists where a malicious DNS seeder eclipses a node by returning an enormous number of IP addresses. In this commit we mitigate this risk by limiting the number of IP addresses addrman learns to 256 per DNS seeder.

GitHub-Pull: #12626
Rebased-From: 46e7f80

l2a5b1 committed:
Fix illegal default `addProxy` and `addrSeparateProxyTor` settings.

GitHub-Pull: #12650
Rebased-From: 40c5886

jnewbery committed:
[config] Remove blockmaxsize option

The blockmaxsize option was marked as deprecated in V0.15.1, and code
was added to convert provided blockmaxsize into blockmaxweight. However,
this code was incorrectly implemented, and blockmaxsize was silently
ignored.

No users have complained about blockmaxsize being ignored, so just
remove it in V0.17.

GitHub-Pull: #12756
Rebased-From: 4757c04

fanquake committed:
depends: Fix Qt build with XCode 9.3

GitHub-Pull: #12946
Rebased-From: 5b4fc3e

MarcoFalke commented on src/init.cpp:
```diff
@@ -489,6 +489,8 @@ std::string HelpMessage(HelpMessageMode mode)
+    if (showDebug)
+        strUsage += HelpMessageOpt("-blockmaxsize=<n>", "Set maximum BIP141 block weight to this * 4. Deprecated, use blockmaxweight");
     strUsage += HelpMessageOpt("-blockmaxweight=<n>", strprintf(_("Set maximum BIP141 block weight (default: %d)"), DEFAULT_BLOCK_MAX_WEIGHT));
     strUsage += HelpMessageOpt("-blockmaxsize=<n>", _("Set maximum BIP141 block weight to this * 4. Deprecated, use blockmaxweight"));
```
This setting is now in here twice.

Also, I think it would make more sense to backport #12756 instead, since it was already "removed" earlier (due to being ignored because of a typo).

MarcoFalke commented:
ACK the gui changes

JeremyRand committed:
Avoid launching as admin when NSIS installer ends.

The Bitcoin Core NSIS script runs with elevated privileges.  Unfortunately, this means that it launches Bitcoin Core itself with elevated privileges when the user chooses to launch Bitcoin Core at the end of the installation procedure.  This commit works around the issue by having explorer.exe launch Bitcoin Core.  Seems to be a similar approach to what http://nsis.sourceforge.net/ShellExecAsUser_plug-in does, but without a plugin.

h/t to "UK" at https://mdb-blog.blogspot.se/2013/01/nsis-lunch-program-as-user-from-uac.html?showComment=1410158039989#c2463780017054126736 for the sample code.

Fixes #7990.

GitHub-Pull: #12985
Rebased-From: 7d8a8cc

sdaftuar commented:
Neither of these are clean cherry-picks, unfortunately, but I thought I'd suggest backporting #12904 and #12902, both of which cause spurious test failures for me (I run the tests on the 0.16 branch regularly).

TheBlueMatt committed:
Default to defining endian-conversion DECLs in compat w/o config

While this isn't a supported build configuration, some build
systems need to build without going through our autotools steps,
so defaulting to something sane may make it easier to build.

Specifically, this fixes the inability to build
rust-bitcoinconsensus on some non-x86 platforms. It needs to build
without our autotools/configure steps to ensure correct compile
args are passed from the rust build system to gcc. Converting the
args from the rust build system to gcc would be a lot of
unmaintainable work.

GitHub-Pull: #12998
Rebased-From: 150b2f0

ken2812221 committed:
qt:Show the entire Window when double clicking on taskbar

GitHub-Pull: #12999
Rebased-From: 67bf2aa

Michagogo commented:
Regarding the NSIS bit, this may not be a good idea: https://mdb-blog.blogspot.com/2013/01/nsis-lunch-program-as-user-from-uac.html?showComment=1388694317801#c939517856791332836

fanquake commented:
@sdaftuar I've added #12902 along with a few other commits. Will have a look at #12904 as well.

sipa committed:
List support for BIP173 in bips.md

fanquake commented:
I've added #12998 as suggested by @TheBlueMatt [here](https://botbot.me/freenode/bitcoin-core-dev/2018-04-24/?msg=99324321&page=2).

laanwj commented:
> Regarding the NSIS bit, this may not be a good idea: https://mdb-blog.blogspot.com/2013/01/nsis-lunch-program-as-user-from-uac.html?showComment=1388694317801#c939517856791332836

Well it's **much** better than launching as admin, even if it is a solution that only works for current windows. In the longer run might be better to remove that option completely.

MarcoFalke commented:
Agree with @laanwj. Will review the diffs before todays meeting

laanwj commented:
utACK 7847b926057b46a5c01e1033f64f6f61dbf3d88f

MarcoFalke commented:
utACK 7847b92605 (Did cherry-picks of all commits that were mentioned in "Rebased-From:" and compared for each that the code diff was zero. Didn't look at the commit messages)

MarcoFalke commented:
@fanquake Could you please manually remove the `blockmaxsize` option in init.cpp, since git's cherry-pick couldn't do that due to a merge conflict?

MarcoFalke commented:
@fanquake Are you still working on this?

fanquake commented:
@MarcoFalke Yes, will cleanup today

On Tue, 1 May 2018 at 03:07, MarcoFalke <notifications@github.com> wrote:

> @fanquake <https://github.com/fanquake> Are you still working on this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/12967#issuecomment-385497970>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA0t8mxV2mw9fVgUrUzTRac5a8WLiaZ2ks5tt2EDgaJpZM4TRh_g>
> .
>

fanquake committed:
Remove blockmaxsize option from init.cpp

MarcoFalke commented:
Thanks. re-ACK 9645aa626230449539423815db6e59ac3cfb57e6

ken2812221 commented:
utACK 8fca086
